woocommerce/woocommerce-android
# Budget screen logic

Part of: #10574 


### Description

This PR adds the following missing logic to the Blaze budget screen: 

- The main budget slider now replicates the behavior of the Blaze webview slider: 
  - Min and Max values are dynamically calculated based on the campaign duration
  - The budget value is updated according to to the duration step, that is, if the duration is 7n days for example, the total budget value will be increased or decreased in steps of 7. This part is trickier than it looks. More on this below. 
- The duration bottom sheet now updates all the budget content once the slider is released, but not while drawing the slider. 
- Implemented updating the campaign start date
- All the data from the Blaze budget screen is now passed back to the campaign preview screen.

The forecast updates are not part of the scope of this PR. 

**About the budget slider**
Due to how the Composable slider works using a Float value to set the slider's position, whereas the budget value is an integer (that has to change in chunks/steps equal to the duration of the campaign), there's some flakiness to how slider responds. Between those 2 values there's some conversion from `float` to `int` and some rounding that causes the total budget value displayed to skip some steps. You'll see what I mean when scrolling back and forth in the budget slider. ~~I tested different approaches (even using the `steps` property from the slider), but none worked perfectly. The current solution is the one that works best. Of course, any ideas for improvement are welcome. Its possible I might be missing something.~~ 

**UPDATE**: About the above. There are some real problems with the current slider implementation from this PR as explained in https://github.com/woocommerce/woocommerce-android/pull/10709#discussion_r1483263929. I fixed them in this other PR: https://github.com/woocommerce/woocommerce-android/pull/10739. 

### Testing instructions

- Open Blaze campaign creation flow
- Open edit budget screen
- Play around with all the sliders and editable info 
- Click `Update` and check that all the update data is reflected in campaign preview screen.

### Images/gif



https://github.com/woocommerce/woocommerce-android/assets/2663464/ef26f6b4-051c-4361-be44-b76532b1523c
